ReactApp/src/utils/rest-api-call.js

```diff
@@ -66,19 +67,47 @@ export async function requestLatexServer(path, body) {
       });
 }
 
+async function fetchWithCookies(path, method, body){
+  var response
+  response = await fetch(SERVER_ROOT_URL + path,
+    {
+        method,
+        headers: {
+            'Content-Type': 'application/json',
+            'Origin': '*',
+            'Authorization': `JWT ${Cookies.get(TOKEN_FIELD_NAME)}`
+        },
+        body: JSON.stringify(body)
+    })
+    return response
+}
+
+async function extractDataFromResponse(response){
+  try{
+    response = await response.json();
+
+    }catch(err){
+        throw new Error("Could not parse response")
+    }
+    
+    if (response.code !== 200) {
+        throw new Error(response.message)
+    }
+    var { code, ...retObject } = response
+    return retObject
+}
+
 export async function requestServer(path, method, body, expectedResponseType) {
     var response
     if (Cookies.get(TOKEN_FIELD_NAME)) {
-        response = await fetch(SERVER_ROOT_URL + path,
-            {
-                method,
-                headers: {
-                    'Content-Type': 'application/json',
-                    'Origin': '*',
-                    'Authorization': `JWT ${Cookies.get(TOKEN_FIELD_NAME)}`
-                },
-                body: JSON.stringify(body)
-            })
+        response = await fetchWithCookies(path,method,body)
+
+        // refresh cookies
+         fetchWithCookies(serverPath.REFRESH_TOKEN ,httpsMethod.GET, undefined).then(async (refreshTokenResponse)=>{
+          const {newToken} = await extractDataFromResponse(refreshTokenResponse)
+          Cookies.set(TOKEN_FIELD_NAME, newToken, {expires: 1 / 96});
```

Comment:
1. It is pretty common to add api to refresh the token, and that is because to send a token in every request is pretty bad as well, it should be depend on time

2. That wont work because there is the res.send() which wont allow for changes to the res after it.

3. The good practice is to change the auth header, but I think that because our client is not a 'restify-client' the header does not received in the react app.

I can try to change the client to restify, but I think that it will may take a lot of time, might not work and we have more urgent things to do

Reply:
ok... 
Is this tested? do you know if it works?

Reply:
90 % it is work. because when I messed up, everything got messed up/

anyway I can check it more properly in the evening.